src/data-access/userRepository.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-unused-vars */
+import { BaseUser, User } from '../types/user.js';
+import { Op } from 'sequelize';
+import { DbError } from '../errors/dbError.js';
+
+
+export class UserRepository {
+    private model;
+    private dataMapper;
+
+    constructor(userModel: any, userDataMapper: any) {
+        this.model = userModel;
+        this.dataMapper = userDataMapper;
+    }
+
+    async get(id: string): Promise<User | undefined> {
+        let userFromDB;
+        try {
+            userFromDB = await this.model.findByPk(id);
+        } catch (e) {
+            throw new DbError('Error retrieving user');
+        }
+        if (userFromDB) {
+            return this.dataMapper.toDomain(userFromDB.toJSON());
```

Comment:
the return should be explicitly used!!!

i.e.

`return userFromDB ? this.dataMapper.toDomain(userFromDB.toJSON()) : null`